sympy/plotting/experimental_lambdify.py

```diff
@@ -107,68 +107,41 @@ class vectorized_lambdify(object):
-                                results.real, copy=False)
-        except Exception as e:
-            #DEBUG: print 'Error', type(e), e
-            if ((isinstance(e, TypeError)
-                 and 'unhashable type: \'numpy.ndarray\'' in str(e))
-                or
-                (isinstance(e, ValueError)
-                 and ('Invalid limits given:' in str(e)
-                      or 'negative dimensions are not allowed' in str(e)  # XXX
-                      or 'sequence too large; must be smaller than 32' in str(e)))):  # XXX
-                # Almost all functions were translated to numpy, but some were
-                # left as sympy functions. They received an ndarray as an
-                # argument and failed.
-                #   sin(ndarray(...)) raises "unhashable type"
-                #   Integral(x, (x, 0, ndarray(...))) raises "Invalid limits"
-                #   other ugly exceptions that are not well understood (marked with XXX)
-                # TODO: Cleanup the ugly special cases marked with xxx above.
-                # Solution: use cmath and vectorize the final lambda.
-                self.lambda_func = experimental_lambdify(
-                    self.args, self.expr, use_python_cmath=True)
-                self.vector_func = np.vectorize(
-                    self.lambda_func, otypes=[np.complex])
-                results = self.vector_func(*args)
-                results = np.ma.masked_where(
-                                np.abs(results.imag) > 1e-7 * np.abs(results),
-                                results.real, copy=False)
-            else:
-                # Complete failure. One last try with no translations, only
-                # wrapping in complex((...).evalf()) and returning the real
-                # part.
-                if self.failure:
-                    raise e
-                else:
-                    self.failure = True
-                    self.lambda_func = experimental_lambdify(
-                        self.args, self.expr, use_evalf=True,
-                        complex_wrap_evalf=True)
-                    self.vector_func = np.vectorize(
-                        self.lambda_func, otypes=[np.complex])
-                    results = self.vector_func(*args)
-                    results = np.ma.masked_where(
-                            np.abs(results.imag) > 1e-7 * np.abs(results),
-                            results.real, copy=False)
-                    warnings.warn('The evaluation of the expression is'
-                            ' problematic. We are trying a failback method'
-                            ' that may still work. Please report this as a bug.')
-        finally:
-            np.seterr(**np_old_err)
-
-        return results
+                np.abs(results.imag) > 1e-7 * np.abs(results),
+                results.real, copy=False)
+            return results
+        except ValueError as e:
+            if self.failure:
+                raise e
```

Comment:
```suggestion
                raise
```
results in simpler tracebacks

Reply:
I've never noticed that difference in the two forms. You'll also have to get rid of the `as e` above to make flake8 happy.